### PR TITLE
On Block Cleanup

### PR DESCRIPTION
**Pull request by nisdas**
**What type of PR is this?**

Cleanup

**What does this PR do? Why is it needed?**

- [x] Cleans up some logic behind our early updates of head to make it simpler to understand.

**Which issues(s) does this PR fix?**

N.A

**Other notes for review**

### CODE REVIEW

File: beacon-chain/blockchain/process_block.go [hunk 1]
```diff
@@ -158,16 +158,10 @@ func (s *Service) onBlock(ctx context.Context, signed *ethpb.SignedBeaconBlock,
 	}
 
 	// Update finalized check point.
-	if (postState.FinalizedCheckpointEpoch() > s.finalizedCheckpt.Epoch) || (featureconfig.Get().UpdateHeadTimely && newFinalized) {
-		if err := s.beaconDB.SaveBlocks(ctx, s.getInitSyncBlocks()); err != nil {
```
Comment (nisdas) [inline beacon-chain/blockchain/process_block.go:162]: This chunk of code is repeated again in `updateFinalized` so we remove it over here.

**Review by rkapka — APPROVED** [review]

File: beacon-chain/blockchain/process_block.go (hunk 1 above)
Comment (farazdagi) [reply beacon-chain/blockchain/process_block.go:162]: Is it ok to ignore the effect of `featureconfig.Get().UpdateHeadTimely` flag? I mean we now have a slightly different semantics: if the first block works (`newFinalized` is true), then this block will also execute, but previously we needed the flag to be on, as well.

Comment (farazdagi) [reply beacon-chain/blockchain/process_block.go:162]: oh, never mind it is a disjunction! so no change in semantics.

Comment (nisdas) [reply beacon-chain/blockchain/process_block.go:162]: The head timely flag is not needed. In the event that it is not applied we still execute the code block due to the first condition. Which basically checks if finalized checkpoint has changed.

**Review by farazdagi — APPROVED** [review]

**Review by terencechain — COMMENTED** [review]
Thanks!

**Pull request merged**